src/local-history/local-history-manager.ts

```diff
@@ -332,4 +337,30 @@ export class LocalHistoryManager {
             vscode.commands.executeCommand('undo');
         }
     }
+
+    /**
+     * Clears all history files which are last modified @param days ago.
+     * @param days The number of days.
+     */
+    public removeOldFiles(days: number): void {
+        const dirPath = path.join(os.homedir(), LOCAL_HISTORY_DIRNAME);
+        if (!fs.existsSync(dirPath)) {
+            return;
+        }
+        const folders = fs.readdirSync(dirPath);
+        for (const folder of folders) {
+            const hashedFolderPath = path.join(os.homedir(), LOCAL_HISTORY_DIRNAME, folder);
+            const files = fs.readdirSync(hashedFolderPath);
+            files.filter((file) => {
+                const currentDate = Date.now();
+                const filePath = path.join(hashedFolderPath, file);
+                const mTime = fs.statSync(filePath).mtimeMs;
+                if (currentDate - mTime > days * dayToMilliSeconds) {
+                    fs.unlinkSync(filePath);
+                }
+            });
+        }
+        vscode.window.showInformationMessage('History successfully deleted');
```

Comment:
Do you think we should display the total count (number of history files deleted) in such a message?

Reply:
Yes that would be nicer something like, `Successfully deleted 32 history files`. I'll update the code.